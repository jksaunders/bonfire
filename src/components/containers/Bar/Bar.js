import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const CONSTANTS = {
  HEIGHT: {
    FULL_VIEW: "100vh"
  }
};

const StyledBar = styled.div`
  height: ${({ height }) => height};
  width: 100%;

  ${props => props.backgroundUrl && `background-image: url('${props.backgroundUrl}');`}
  ${props => props.backgroundUrl && "background-size: 100%;"}
`;

const propTypes = {
  backgroundUrl: PropTypes.string,
  children: PropTypes.node,
  height: PropTypes.string,
  parallax: PropTypes.bool,
};

const defaultProps = {
  backgroundUrl: null,
  children: null,
  height: "300px",
  parallax: false
};

const Bar = ({
  children,
  ...rest
}) => (
  <StyledBar {...rest}>
    {children}
  </StyledBar>
);

Bar.propTypes = propTypes;
Bar.defaultProps = defaultProps;
Bar.CONSTANTS = CONSTANTS;

export default Bar;