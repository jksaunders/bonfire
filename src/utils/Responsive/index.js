export { default as ResponsiveRoot } from './ResponsiveRoot';
export * from './ResponsiveRoot';
