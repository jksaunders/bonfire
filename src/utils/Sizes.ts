export type SizeProp<T> = Record<string, T>;
