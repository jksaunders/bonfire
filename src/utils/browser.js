const goTo = (url) => {
  if (url && typeof window !== 'undefined') {
    window.location.href = url;
  }
};

export default {
  goTo
};