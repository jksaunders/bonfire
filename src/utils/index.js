export { default as Browser } from "./browser";
export * from "./styling";