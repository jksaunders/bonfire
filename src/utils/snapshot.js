/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";

export const expectSnapshot = (mountedComponent) => {
  const component = renderer.create(mountedComponent);
  expect(component.toJSON()).toMatchSnapshot();
};