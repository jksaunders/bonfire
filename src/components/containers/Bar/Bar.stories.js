import React from 'react';
import styled, { ThemeProvider } from 'styled-components';

import { select, withKnobs } from '@storybook/addon-knobs';

import { ThemeConstants } from '../../../theming';
import Bar from './Bar';
import Typography, { MaterialVariants } from '../../Typography';

const StoryWrapper = styled.div``;

const StyledBar = styled(Bar)`
  align-items: center;
  display: flex;
  justify-content: center;
`;

const Content = styled.div`
  background-color: rgba(255, 255, 255, 0.7);
  padding: 32px;
`;

export default {
  title: 'Containers|Bar',
  component: Bar,
  decorators: [withKnobs],
};

export const BarStory = () => {
  const themeMode = select(
    'Theme Mode',
    Object.keys(ThemeConstants.mode.values),
    Object.keys(ThemeConstants.mode.values)[0]
  );
  const themeLayout = select(
    'Theme Layout',
    Object.keys(ThemeConstants.layout.values),
    Object.keys(ThemeConstants.layout.values)[0]
  );
  const theme = { mode: themeMode, layout: themeLayout };

  return (
    <ThemeProvider theme={theme}>
      <StoryWrapper>
        <StyledBar backgroundUrl="https://images.unsplash.com/photo-1497906539264-eb74442e37a9?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1868&q=80">
          <Content>
            <Typography variant={MaterialVariants.H2}>bar content</Typography>
          </Content>
        </StyledBar>
        <StyledBar>
          <Content>
            <Typography variant={MaterialVariants.H2}>bar content</Typography>
          </Content>
        </StyledBar>
        <StyledBar backgroundUrl="https://images.unsplash.com/photo-1497906539264-eb74442e37a9?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1868&q=80">
          <Content>
            <Typography variant={MaterialVariants.H2}>bar content</Typography>
          </Content>
        </StyledBar>
      </StoryWrapper>
    </ThemeProvider>
  );
};

BarStory.story = {
  name: 'with text',
};
