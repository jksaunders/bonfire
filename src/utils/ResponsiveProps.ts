import { SizeProp } from './Sizes';

export type ResponsiveProp<T> = T | SizeProp<T>;
export type ResponsiveProps<T, ResponsiveKeys = keyof T> = {
  [K in keyof T]: K extends ResponsiveKeys
    ? T[K] extends undefined
      ? ResponsiveProp<T[K]>
      : ResponsiveProp<NonNullable<T[K]>>
    : T[K];
};
