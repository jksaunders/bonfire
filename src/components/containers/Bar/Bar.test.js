import React from "react";
import { mount } from "enzyme";
import { expectSnapshot } from "../../../utils/snapshot";
import Bar from "./Bar";

const getBar = (props = {}) => <Bar {...props} />;

test("Bar mounts", () => {
  const wrapper = mount(getBar());
  expect(wrapper).toExist();
  expectSnapshot(getBar());
});