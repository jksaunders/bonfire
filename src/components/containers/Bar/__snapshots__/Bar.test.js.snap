// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar mounts 1`] = `
<div
  className="sc-bdVaJa gKjoUk"
  height="300px"
/>
`;
